ext/afform/core/Civi/Afform/Translator.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Civi\Afform;
+
+use Civi\Core\Service\AutoService;
+use Smarty\Filter\FilterInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * @service civi.afform.translator
+ */
+class Translator extends AutoService implements EventSubscriberInterface {
+
+  public static function getSubscribedEvents() {
+    return [
+      '&hook_civicrm_alterAngular' => 'translateAfform',
+    ];
+  }
+
+  /**
+   * @param \Civi\Angular\Manager $angular
+   */
+  public function translateAfform($angular) {
+    // 1. only formBuilder = ok
+    // 2. can we make it work before some other angular complexity is added (e.g. icons) = faux problème 
+    // 3. loop on selectors
+
+/*
+- af-entity[label]
+- af-field[defn.label]
+- af-field[defn.help_pre]
+- af-field[defn.help_post]
+- af-field[defn.placeholder]
+- af-field[defn.options.label] => (Custom options for Date -- Or we have a custom case for all date fields) i.e. defn="{options: [{id: 'this.day', label: 'Today'}
+- p.af-text (should we add a class for saying it's translatable?)
+- div.af-markup (should we add a class for saying it's translatable?) -> it should be assumed to be 
+- fieldset[af-copy] (label of the copie button)
+- fieldset[af-repeat] (label of the repeat button)
+- button
+- [af-title] */
+
+    $changeSet = \Civi\Angular\ChangeSet::create('translate')
+      ->alterHtml(';.aff.html;', function (\phpQueryObject $doc) {
```

Comment:
Tighter regex?

```suggestion
      ->alterHtml(';\.aff\.html$;', function (\phpQueryObject $doc) {
```

Reply:
Thanks @totten this is fixed as suggested.